specification/dsdl/architecture.tex

```diff
@@ -193,27 +189,26 @@ \subsection{File hierarchy}
 
 \subsection{Elements of data type definition}
 
-A data type definition file contains an exhaustive description of a particular version of the said data type in the
+A data type definition file contains an exhaustive description of a particular version of the data type in the
 \emph{data structure description language} (DSDL).
-As explained above, one data type definition contains either one or two data schema definitions,
-for message types and service types, respectively.
+As explained above, one data type definition contains one schema definition for message types or two schema definitions for service types.
 
 A data schema definition contains an ordered, possibly empty collection of \emph{field attributes} and/or
 unordered, possibly empty collection of \emph{constant attributes}.
 A data schema may describe either a \emph{structure object} or a \emph{tagged union object}.
 The value of a structure object is a function of the values of all of its field attributes.
 A tagged union object is formed from at least two field attributes,
 but it is capable of holding exactly one field attribute value at any given time.
-The value of a tagged union object is a function of which field attribute value
-it is holding at the moment and the value of said field attribute.
+The value of a tagged union object is a function of the field attribute values
+it is presently holding and the value of these field attributes.
 
 A field attribute represents a named dynamically assigned value of a statically defined type
 that can be exchanged over the network as a member of its containing object.
 A padding field attribute is a special kind of field attribute which is used for data alignment purposes;
 such field attributes are not named.
 
 A constant attribute represents a named statically defined value of a statically defined type.
-Constants are never exchanged over the network, since they are assumed to be known to all involved nodes
+Constants are never exchanged over the network, since they are assumed to all involve nodes
```

Comment:
The point here is that if all nodes share the definition of a data type, they all have access to the constants defined therein, so there is no point exchanging their definitions over the network.

Reply:
Also reverting.